Allow users to supply a Slack app id and secret.

jonthegeek opened the pull request:
I'm still getting tests sorted out, but this should work. This way users can authenticate using their own Slack apps, and thus can use this for logging into their own sites (with their own redirect URLs; right now we're restricted to localhost or the r4ds shinyapps account).

jonthegeek committed:
Allow users to supply a Slack app id and secret.

I'm still getting tests sorted out, but this should work. This way users can authenticate using their own Slack apps, and thus can use this for logging into their own sites (with their own redirect URLs; right now we're restricted to localhost or the r4ds shinyapps account).

yonicd commented on R/team_managment.R:
```diff
@@ -35,8 +36,13 @@ add_team <- function(team, token) {
+                           verbose = TRUE,
+                           client_id = NULL,
+                           client_secret = NULL) {
+  client_id <- client_id %||% builtin_client_id
```
why is this needed here? it isnt used in the function. same line 45

yonicd replied:
cleaned up in new commit

Commit:
options implementation of token and secret control

Commit:
remove builtin prefix from sysdata

jonthegeek committed:
Format docs.

jonthegeek committed:
Apply fix to sysdata.

jonthegeek commented:
Converted back to draft. I want to add a vignette walking through the process of supplying your own client (and walk through that process myself to make sure it works as expected; it's been a long time since I created a new client).

jonthegeek committed:
Information about using custome Slack Apps.

jonthegeek commented:
Tweaking the vignette.

jonthegeek committed:
Close the loop on the vignette.

Also cleaned some things up, and switched the custom app info to environment variables for easier use.

yonicd commented:
not sure why i thought formal in `Sys.getenv` was `default`

jonthegeek commented:
> not sure why i thought formal in `Sys.getenv` was `default`

You had them as getOption, which *does* use `default`. I switched it to `Sys.getenv`, because it's easier/more likely for a user to set these things through an environment variable (eg, `.Renviron`, which shinyapps.io recognizes) rather than an option.
